legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabRevisionApi.java

```diff
@@ -147,11 +150,12 @@ private RevisionAccessContext getWorkspaceEntityRevisionContextByWorkspaceAccess
         ProjectFileAccessProvider.FileAccessContext fileAccessContext = fileAccessProvider.getFileAccessContext(projectId, workspaceId, workspaceAccessType, null);
         ProjectStructure projectStructure = ProjectStructure.getProjectStructure(fileAccessContext);
         String filePath = projectStructure.findEntityFile(entityPath, fileAccessContext);
-        if (filePath == null)
+        String canonicalizedFilePath = ProjectPaths.canonicalizeFile(filePath);
+        if (canonicalizedFilePath == null)
```

Comment:
This check should be done on filePath and prior to canonicalizing. The call to findEntityFile is what actually checks that the entity file exists.